iai-callgrind-runner/src/runner/bin_bench.rs

```diff
@@ -873,3 +900,343 @@ impl Benchmark for SaveBaselineBenchmark {
 pub fn run(benchmark_groups: BinaryBenchmarkGroups, config: Config) -> Result<()> {
     Runner::new(benchmark_groups, config)?.run()
 }
+
+impl Delay {
+    fn defaults(mut self) -> Self {
+        match self.kind {
+            DelayKind::DurationElapse(_) => {
+                if self.poll.is_some() {
+                    warn!("Ignoring poll setting. Not support for {:?}", self.kind);
+                    self.poll = None;
+                }
+                if self.timeout.is_some() {
+                    warn!("Ignoring timeout setting. Not support for {:?}", self.kind);
+                    self.timeout = None;
+                }
+            }
+            DelayKind::PathExists(_) | DelayKind::TcpConnect(_) | DelayKind::UdpResponse(_, _) => {
+                if self.timeout.is_none() {
+                    self.timeout = Some(Duration::from_secs(600));
+                };
+                if self.poll.is_none() {
+                    self.poll = Some(Duration::from_millis(10));
+                };
+            }
+        }
+        if let (Some(poll), Some(timeout)) = (self.poll, self.timeout) {
+            if poll >= timeout {
```

Comment:
Changed the behaviour to set `poll` to `timeout -5ms` in case `timeout >= 5ms`, otherwise using `timeout`.
Updated the documentation for `Command::poll` & `Command::timeout`.